### PR TITLE
Remove name verification

### PR DESCRIPTION
**Pull request by tt**
This replaces #1120.

### CODE REVIEW

File: registry/storage/manifeststore.go [hunk 1]
```diff
@@ -106,9 +106,6 @@ func (ms *manifestStore) GetByTag(tag string, options ...distribution.ManifestSe
 // content, leaving trust policies of that content up to consumers.
 func (ms *manifestStore) verifyManifest(ctx context.Context, mnfst *schema1.SignedManifest) error {
 	var errs distribution.ErrManifestVerification
-	if mnfst.Name != ms.repository.Name() {
```
Comment (stevvooe) [inline registry/storage/manifeststore.go]: I think we still need to validate the name, but we don't need to match it to the repo.

Comment (tt) [inline registry/storage/manifeststore.go]: Validate how? I thought this would remove only the matching.

Comment (stevvooe) [inline registry/storage/manifeststore.go]: I think it is sufficient to validate against `reference.NameRegexp`.

**Comment by endophage** [comment]
With some of the recent problems we've seen, can we confirm that the docker client/engine doesn't care about the name contained in a manifest before we remove server side verification?

If the client verifies the name then this PR should not proceed. Sorry for getting in here so late, we should not be weakening data integrity checks that allow users to push data a client will never accept on pull. If clients don't care about the name at all then this seems fine.

**Comment by tt** [comment]
@endophage, see [this comment](https://github.com/docker/distribution/pull/1120#issuecomment-156535838) by @dmcgowan.

**Comment by stevvooe** [comment]
LGTM

Errors may need to be mapped correctly to avoid 500s in webapp.

**Comment by dmcgowan** [comment]
+1 for using a different error, `fmt.Errorf("invalid manifest name format")`  could be defined and mapped.

otherwise LGTM

**Comment by endophage** [comment]
@tt thanks, sounds good!

**Comment by tt** [comment]
Mapped how, @stevvooe?

I don't see [these errors](https://github.com/docker/distribution/blob/073bd8926224a9a4cc7c7fc90fdd3828046c381b/reference/reference.go#L42-L58) referenced outside the package that defines them.

**Comment by stevvooe** [comment]
@tt The errors get added to a https://godoc.org/github.com/docker/distribution#ErrManifestVerification struct. These get mapped in the manifest handler [here](https://github.com/docker/distribution/blob/master/registry/handlers/images.go#L167). New types can be added to represent the condition. Otherwise, these will be treated as 500s. These should be `ErrManifestXXX` errors, namewise.

**Comment by tt** [comment]
@stevvooe, defined `ErrManifestNameInvalid` which I mapped to `ErrorCodeNameInvalid`. The latter was already defined but this error seems to be a case of what it should probably represent.

**Comment by stevvooe** [comment]
LGTM

**Comment by dmcgowan** [comment]
LGTM

**Comment by stevvooe** [comment]
@tt Thanks for working with us to get this through!

**Pull request merged**